scripts/cleanup_old_vms.py

```diff
@@ -217,75 +222,87 @@ def cleanup_vms(texts, max_hours=24, providers=None, prompt=True):
     Prompts user to continue with delete
     Threads deleting of the vms
 
-    :param texts: list of strings to match against
-    :param max_hours: integer maximum number of hours that the VM can exist for
-    :param providers: list of provider keys
-    :param prompt: boolean, whether or not to prompt the user for each delete
-    :return: 0 if user declines delete when prompt is True
+    Args:
+        texts (list): List of regex strings to match with
+        max_hours (int): age limit for deletion
+        providers (list): List of provider keys to scan and cleanup
+        prompt (bool): Whether or not to prompt the user before deleting vms
+    Returns:
+        int: return code, 0 on success, otherwise raises exception
     """
     providers = providers or list_provider_keys()
     delta = datetime.timedelta(hours=int(max_hours))
-    vms_to_delete = defaultdict(set)
-    thread_queue = []
-    # precompile regexes
-    matchers = [re.compile(text, re.IGNORECASE) for text in texts]
+    # Compile regex, strip leading/trailing single quotes from cli arg
+    matchers = [re.compile(text.strip("'"), re.IGNORECASE) for text in texts]
     print('Matching VM names against the following case-insensitive strings: {}'.format(texts))
 
+    providers_to_scan = []
     for provider_key in providers:
         # check for cleanup boolean
         if not cfme_data['management_systems'][provider_key].get('cleanup', False):
             print('Skipping {}, cleanup map set to false or missing in yaml'.format(provider_key))
             continue
+        # check the provider is reachable
         ipaddress = cfme_data['management_systems'][provider_key].get('ipaddress')
         if ipaddress and not net.is_pingable(ipaddress):
             continue
-        thread = Thread(target=process_provider_vms,
-                        args=(provider_key, matchers, delta, vms_to_delete))
-        # Mark as daemon thread for easy-mode KeyboardInterrupt handling
-        thread.daemon = True
-        thread_queue.append(thread)
-        thread.start()
-
-    # Join the queued calls
-    for thread in thread_queue:
-        thread.join()
+        # passed the checks
+        providers_to_scan.append(provider_key)
+
+    scan_queue = Queue()
+    scan_failure_queue = Queue()
+    scan_proc_list = []
+    for provider_key in providers_to_scan:
+        scan_proc_list.append(
+            Process(target=scan_provider,
+                    args=(provider_key, matchers, delta, scan_queue, scan_failure_queue)))
+
+    for proc in scan_proc_list:
+        proc.start()
+    for proc in scan_proc_list:
+        proc.join()
```

Comment:
ah gotcha - absolutely, and that's the intent, to wait until all the scans are done to process the Queue of the matching VM's